templates/crd-servicerouters.yaml

```diff
@@ -72,18 +77,74 @@ spec:
                             the request when a retryable result occurs
                           format: int32
                           type: integer
+                        partition:
+                          description: Partition is the Consul partition to resolve
+                            the service from instead of the current partition. If
+                            empty the current partition is assumed.
+                          type: string
                         prefixRewrite:
                           description: PrefixRewrite defines how to rewrite the HTTP
                             request path before proxying it to its final destination.
                             This requires that either match.http.pathPrefix or match.http.pathExact
                             be configured on this route.
                           type: string
+                        requestHeaders:
+                          description: Allow HTTP header manipulation to be configured.
+                          properties:
+                            add:
+                              additionalProperties:
+                                type: string
+                              description: Add is a set of name -> value pairs that
+                                should be appended to the request or response (i.e.
+                                allowing duplicates if the same header already exists).
+                              type: object
+                            remove:
+                              description: Remove is the set of header names that
+                                should be stripped from the request or response.
+                              items:
+                                type: string
+                              type: array
+                            set:
+                              additionalProperties:
+                                type: string
+                              description: Set is a set of name -> value pairs that
+                                should be added to the request or response, overwriting
+                                any existing header values of the same name.
+                              type: object
+                          type: object
                         requestTimeout:
                           description: RequestTimeout is the total amount of time
                             permitted for the entire downstream request (and retries)
                             to be processed.
-                          format: int64
```

Comment:
Hmm. This looks like we did a breaking change between consul-k8s versions?

Reply:
correct.. but the previous one wouldnt have worked as a duration, so in some sense, we are actually ensuring the user can supple duration values and have them behave as such. imo if they were using this, they would have called us out on the CRD failing when they set "10s" for example.